src/OrchardCore.Modules/OrchardCore.Media/Services/MediaCacheBackgroundTask.cs

```diff
@@ -0,0 +1,160 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Logging;
+using OrchardCore.BackgroundTasks;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Media.Core;
+
+namespace OrchardCore.Media.Services
+{
+
+    // At 12:00 on Monday.
+    [BackgroundTask(Schedule = "0 12 * * 1", Description = "Performs cleanup operations for ms-cache and is-cache folders periodically.")]
+    // Left for test driving.
+    //[BackgroundTask(Schedule = "* * * * *", Description = "Performs cleanup operations for ms-cache and is-cache folders periodically.")]
+    public class MediaCacheBackgroundTask : IBackgroundTask
+    {
+        private readonly IWebHostEnvironment _webHostEnvironment;
+        private readonly ShellSettings _shellSettings;
+        private readonly ILogger _logger;
+        private const int Repeats = 5;
+        private const int RepeatTime = 5000;
+
+        public MediaCacheBackgroundTask(
+        IWebHostEnvironment webHostEnvironment,
+        ShellSettings shellSettings,
+        ILogger<MediaCacheBackgroundTask> logger)
+        {
+            _webHostEnvironment = webHostEnvironment;
+            _shellSettings = shellSettings;
+            _logger = logger;
+        }
+
+        public Task DoWorkAsync(IServiceProvider serviceProvider, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Media cache background task cleaning started");
+
+            var directoryInfo = new DirectoryInfo(Path.Combine(_webHostEnvironment.WebRootPath, "is-cache"));
+
+            // Don't delete is-cache folder.
+            RecursiveDeleteAsync(directoryInfo, false, cancellationToken);
+
+            directoryInfo = new DirectoryInfo(GetMediaCachePath(_webHostEnvironment, DefaultMediaFileStoreCacheFileProvider.AssetsCachePath, _shellSettings));
+            RecursiveDeleteAsync(directoryInfo, false, cancellationToken);
```

Comment:
@Skrypt 

If the idea is to purge-all, you should use `IMediaFileStoreCacheFileProvider` and call `PurgeAll()` method which does the same thing.

Reply:
I would have used that method if that would have worked all the time. The thing I'm trying to do here is to find out why it fails to delete some of these files because of a lock. I don't want to use another service that does that because I don't want to change the service itself. Once, we will have figured out the lock then we will update that service accordingly.